core/src/main/java/hudson/model/Run.java

```diff
@@ -413,9 +413,14 @@ public void addAction(@Nonnull Action a) {
 
     /**
      * Ordering based on build numbers.
+     * If numbers are equal order based on names of parent projects.
      */
     public int compareTo(@Nonnull RunT that) {
-        return this.number - that.number;
+        final int res = this.number - that.number;
+        if (res == 0)
+            return this.getParent().getFullName().compareTo(that.getParent().getFullName());
```

Comment:
Do we really need to compare full names in this case? Maybe just a comparison of instances is enough (though there may be corner cases in such case)